src/Platform/Microsoft.Testing.Platform/Hosts/ServerTestHost.cs

```diff
@@ -553,7 +552,8 @@ await ExecuteRequestAsync(
                     testNodeUpdateProcessor.GetTestNodeStatistics().TotalDiscoveredTests)
                 : throw new NotImplementedException($"Request not implemented '{method}'");
 
-        await _telemetryService.LogEventAsync(TelemetryEvents.TestsRunEventName, metadata).ConfigureAwait(false);
+        // TODO: Should this be ServiceProvider.GetTestApplicationCancellationTokenSource()? Or perRequestTestSessionContext.CancellationToken?
```

Comment:
Not sure here.

Reply:
Telemetry and update I think it should be "the global one" the per session only to "cancel single call"